File: app/src/main/java/com/wayfair/themoviedb/feature/searchresults/SearchResultsScreenModule.java
package com.wayfair.themoviedb.feature.searchresults;

import dagger.Binds;
import dagger.Module;

@Module
public abstract class SearchResultsScreenModule {
    @Binds
    abstract SearchResultsContract.Presenter providesSearchResultsPresenter(SearchResultsPresenter presenter);

    @Binds
    abstract SearchResultsContract.Interactor providesSearchResultsInteractor(SearchResultsInteractor interactor);

    @Binds
    abstract SearchResultsContract.Repository providesSearchResultsRepository(SearchResultsRepository repository);

    @Binds
    abstract SearchResultsContract.Router providesSearchResultsRouter(SearchResultsRouter router);
}

File: app/src/main/java/com/wayfair/themoviedb/feature/searchresults/SearchResultsScreenModule.kt
package com.wayfair.themoviedb.feature.searchresults

import com.wayfair.themoviedb.di.ScreenScope
import dagger.Binds
import dagger.Module
import dagger.Provides

@Module
abstract class SearchResultsScreenModule {
    @Binds
    abstract fun providesSearchResultsPresenter(presenter: SearchResultsPresenter): SearchResultsContract.Presenter

    @Binds
    abstract fun providesSearchResultsInteractor(interactor: SearchResultsInteractor): SearchResultsContract.Interactor

    @Binds
    abstract fun providesSearchResultsRepository(repository: SearchResultsRepository): SearchResultsContract.Repository

    @Binds
    abstract fun providesSearchResultsRouter(router: SearchResultsRouter): SearchResultsContract.Router

    @Module
    companion object {
        @JvmStatic
        @ScreenScope
        @Provides
        fun providesContext(fragment: SearchResultsFragment) = fragment.resources
    }
}
